src/main/java/net/mcreator/element/types/Achievement.java

```diff
@@ -96,4 +102,24 @@ public boolean hasRewards() {
 		};
 	}
 
+	@Override public Collection<String> getUsedElementNames() {
+		List<String> elements = new ArrayList<>();
+		if (rewardFunction != null && !rewardFunction.equals("No function"))
+			elements.add("CUSTOM:" + rewardFunction);
+		rewardLoot.forEach(e -> elements.add("CUSTOM:" + e));
+		rewardRecipes.forEach(e -> elements.add("CUSTOM:" + e));
+		return elements;
+	}
+
+	@Override public Collection<? extends MappableElement> getUsedElementMappings() {

```

Comment:
This would be one example. (no parent) is non-existing mapping, meaning this ME does not reference any MappableElement for the parent.

Probably other MEs need a change in logic similar to this.

All reported references in all MEs need to be valid, list can not contain references that do not actually exist or are not present in the generated code.

Reply:
✔️ Fixed, only advancement and tab entries seemed to need such handling.